### ISSUE TITLE
per-dir client certs or ciphers

### ISSUE THREAD

**Issue by covener** [abetterinternet/mod_tls#7]
Hi, take with a grain of salt as I primarily work with a proprietary SSL module but one thing mod_ssl has that "my" module doesn't is per-directory TLS client certificate verification or TLS cipher renegotiation

For client certs, we get a lot of requests to simulate it with "optional" + e.g. mod_rewrite.
For ciphers, it's a less popular question but there is not a great solution (accept weak ciphers then redirect to an error page). I don't know if/how it works in mod_ssl with tls13.

From a glance at tls_conf.c it seems like mod_tls has the same gap? It might be worth a mention in the feature comparison or just as an issue in the backlog here.

**Comment by icing** [comment]
Yeah, client certs need some more infrastructure on the `rustls` side for inspection and revocation, mainly.

As to directory based settings, that is left out, mainly as TLS1.3 does not support renegotiation to my understanding and we would create configurations that work with v1.2 only.

This is one of the reasons why I wanted a design in the server that allows SSL modules co-exist. At least in front-/backend split or on separate ports.

The case with the redirect on weak ciphers is interesting. Never thought about hat.

I agree that this should be mentioned in the comparision matrix.
